### PR TITLE
Adds FUSD to Kitty Items

### PR DESCRIPTION
**Pull request by 10thfloor**

### CODE REVIEW

File: web/src/flow/tx.create-sale-offer.js [hunk 1]
```diff
@@ -5,22 +5,23 @@ import {tx} from "./util/tx"
...
   import NonFungibleToken from 0xNonFungibleToken
-  import Kibble from 0xKibble
+  import FUSD from 0xFUSD
   import KittyItems from 0xKittyItems
   import NFTStorefront from 0xNFTStorefront
 
   transaction(saleItemID: UInt64, saleItemPrice: UFix64) {
 
-    let kibbleReceiver: Capability<&Kibble.Vault{FungibleToken.Receiver}>
+    let FUSDReceiver: Capability<&FUSD.Vault{FungibleToken.Receiver}>
```
Comment (psiemens) [inline web/src/flow/tx.create-sale-offer.js:14]:
```suggestion
    let fusdReceiver: Capability<&FUSD.Vault{FungibleToken.Receiver}>
```

File: web/src/flow/tx.create-sale-offer.js [hunk 2]
```diff
@@ -5,22 +5,23 @@ import {tx} from "./util/tx"
...
     let kittyItemsCollection: Capability<&KittyItems.Collection{NonFungibleToken.Provider, NonFungibleToken.CollectionPublic}>
     let storefront: &NFTStorefront.Storefront
 
     prepare(account: AuthAccount) {
       // We need a provider capability, but one is not provided by default so we create one if needed.
       let kittyItemsCollectionProviderPrivatePath = /private/kittyItemsCollectionProvider
 
-      self.kibbleReceiver = account.getCapability<&Kibble.Vault{FungibleToken.Receiver}>(Kibble.ReceiverPublicPath)!
-      assert(self.kibbleReceiver.borrow() != nil, message: "Missing or mis-typed Kibble receiver")
+      self.FUSDReceiver = account.getCapability<&FUSD.Vault{FungibleToken.Receiver}>(/public/fusdReceiver)!
```
Comment (psiemens) [inline web/src/flow/tx.create-sale-offer.js:22]:
```suggestion
      self.fusdReceiver = account.getCapability<&FUSD.Vault{FungibleToken.Receiver}>(/public/fusdReceiver)!
```

File: web/src/flow/tx.create-sale-offer.js [hunk 3]
```diff
@@ -5,22 +5,23 @@ import {tx} from "./util/tx"
...
+
+      assert(self.FUSDReceiver.borrow() != nil, message: "Missing or mis-typed FUSD receiver")
```
Comment (psiemens) [inline web/src/flow/tx.create-sale-offer.js:24]:
```suggestion
      assert(self.fusdReceiver.borrow() != nil, message: "Missing or mis-typed FUSD receiver")
```

File: web/src/flow/tx.create-sale-offer.js [hunk 4]
```diff
@@ -35,15 +36,15 @@ const CODE = fcl.cdc`
 
     execute {
       let saleCut = NFTStorefront.SaleCut(
-        receiver: self.kibbleReceiver,
+        receiver: self.FUSDReceiver,
```
Comment (psiemens) [inline web/src/flow/tx.create-sale-offer.js:39]:
```suggestion
        receiver: self.fusdReceiver,
```

File: web/src/parts/balance-cluster.comp.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import {Suspense} from "react"
 import {useFlowBalance} from "../hooks/use-flow-balance.hook"
-import {useKibblesBalance} from "../hooks/use-kibbles-balance.hook"
+import {useFUSDBalance} from "../hooks/use-fusd-balance.hook"
 import {useCurrentUser} from "../hooks/use-current-user.hook"
 import {IDLE} from "../global/constants"
-import {fmtKibbles} from "../util/fmt-kibbles"
+import {fmtKibbles} from "../util/fmt-fusd"
```
Comment (psiemens) [inline web/src/parts/balance-cluster.comp.js:6]:
```suggestion
import {fmtFUSD} from "../util/fmt-fusd"
```

File: web/src/parts/balance-cluster.comp.js [hunk 6]
```diff
@@ -40,9 +40,9 @@ export function BalanceCluster({address}) {
         <Table size="sm">
           <Tbody>
             <Tr>
-              <Td>KIBBLE</Td>
-              {kibbles.status === IDLE ? (
-                <Td isNumeric>{fmtKibbles(kibbles.balance)}</Td>
+              <Td>FUSD</Td>
+              {fusd.status === IDLE ? (
+                <Td isNumeric>{fmtKibbles(fusd.balance)}</Td>
```
Comment (psiemens) [inline web/src/parts/balance-cluster.comp.js:45]:
```suggestion
                <Td isNumeric>{fmtFUSD(fusd.balance)}</Td>
```

File: web/src/parts/kibbles-balance-cluster.comp.js [hunk 7]
```diff
@@ -1,29 +1,29 @@
 import {Suspense} from "react"
 import {useInitialized} from "../hooks/use-initialized.hook"
-import {useKibblesBalance} from "../hooks/use-kibbles-balance.hook"
+import {useFUSDBalance} from "../hooks/use-fusd-balance.hook"
 import {Bar, Label, Button} from "../display/bar.comp"
 import {IDLE} from "../global/constants"
 import {Loading} from "../parts/loading.comp"
-import {fmtKibbles} from "../util/fmt-kibbles"
+import {fmtKibbles} from "../util/fmt-fusd"
```
Comment (psiemens) [inline web/src/parts/kibbles-balance-cluster.comp.js:7]:
```suggestion
import {fmtFUSD} from "../util/fmt-fusd"
```

File: api/src/routes/fusd.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    });
+  });
+
+  router.post(
+    "/fusd/transfer",
+    [
+      body("recipient").exists(),
+      body("amount").isInt({
+        gt: 0
+      })
```
Comment (psiemens) [inline api/src/routes/fusd.ts:36]: I think this should be `isDecimal`

File: api/src/config.ts [hunk 9]
```diff
@@ -29,6 +29,8 @@ export function getConfig(env) {
 
   const nonFungibleTokenAddress = env.NON_FUNGIBLE_TOKEN_ADDRESS!;
 
+  const fusdAddress = env.FUSD_TOKEN_ADDRESS! || env.MINTER_ADDRESS!;
```
Comment (psiemens) [inline api/src/config.ts:32]:
```suggestion
  const fusdAddress = env.FUSD_ADDRESS! || env.MINTER_ADDRESS!;
```

File: .env.testnet [hunk 10]
```diff
@@ -2,6 +2,7 @@ CHAIN_ENV=testnet
 
 FUNGIBLE_TOKEN_ADDRESS=0x9a0766d93b6608b7
 NON_FUNGIBLE_TOKEN_ADDRESS=0x631e88ae7f1d7c20
+FUSD_TOKEN_ADDRESS=0xe223d8a629e49c68
```
Comment (psiemens) [inline .env.testnet:5]:
```suggestion
FUSD_ADDRESS=0xe223d8a629e49c68
```

Just to be consistent with the `FUSD_ADDRESS` variables below

File: api/.env.local [hunk 11]
```diff
@@ -4,6 +4,7 @@ MINTER_ACCOUNT_KEY_INDEX=0
 
 FUNGIBLE_TOKEN_ADDRESS=${FUNGIBLE_TOKEN_ADDRESS}
 NON_FUNGIBLE_TOKEN_ADDRESS=${NON_FUNGIBLE_TOKEN_ADDRESS}
+FUSD_TOKEN_ADDRESS=${FUSD_ADDRESS}
```
Comment (psiemens) [inline api/.env.local:7]: Oh, is this to avoid a circular variable reference?

Is this `api/.env.local` file still needed? 🤔

**Review by psiemens — COMMENTED** [review]
🎉 🎉

File: api/.env.local (hunk 11 above)
Comment (10thfloor) [reply api/.env.local:7]: Hmm, yeah _maybe_ not.

File: api/src/routes/fusd.ts [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+    return res.send({
+      transaction
+    });
+  });
+
+  router.post(
+    "/fusd/transfer",
+    [
+      body("recipient").exists(),
+      body("amount").isInt({
```
Comment (10thfloor) [inline api/src/routes/fusd.ts:34]:
```suggestion
      body("amount").isDecimal({
```

**Pull request merged**